docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -32,197 +35,203 @@ note over ERCC1, ERCC2 #FFaa00
      be dependent on such policy.
 end note
 
-group enclave creation
-  activate Admin
-
-  note over Admin #FFaa00
-    These operations are assumed to be performed only after the chaincode definition is
-    committed on the channel for two reasons:
-    1) chaincode invocation may not be available before the definition is committed,
-    2) enclave registration should be feasible only for committed chaincodes
-
-    PS: in a future tighter integration with fabric we could remove the first constraint
-    and could potentially piggyback the steps before the ercc registration (enclave and
-    evidence generation) in the `approveformyorg` and the registration as part of
-    `commit` without having to change the protocol flows.
-  end note
+Admin -> Peer_CLI ++: lifecycle chaincode\n createenclave (CC_Id)
 
-  note over Admin, E_Manager #FFaa00
-    The E_Manager component hides some operations that do not necessarily need to be
-    performed by the Admin -- they could be scripted or embedded in the peer process;
-    depending on the actual implementation it may require admin credential on the peer.
+group enclave creation
```

Comment:
Right now we do external builders only in invocation and handle there the run.   Effectively, though, the run will be triggered here!  Note we refer later to external builder for storage, so the diagram is somewhat external builder aware.

Maybe just add a forward reference to the invocation part?  Or, just for our own benefit, add this observation as a puml comment, so we remember?

We also don't handle the detect and build external builder commands which would have to be in the lifecycle diagram. But that i guess seems fine, the crucial part of the external builder is really interaction of run, it's scratchspace in the filesystem and invocations ....

Reply:
Ok for the forward reference -- it adds clarity.
Not sure documenting the external builder adds anything; perhaps, as you say, the run part, particularly for handling sealed storage.

Reply:
yep, the run part is useful because of the sealing and is done already in fpc-invocation.  For the other part i also think it's not essential (otherwise i would already have commented in the fpc-invocation PR :-)